lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -352,6 +356,102 @@ func (c *ContainerProxy) ContainerRunning(vc *viccontainer.VicContainer) (bool,
 	return inspectJSON.State.Running, nil
 }
 
+func (c *ContainerProxy) Wait(vc *viccontainer.VicContainer, timeout time.Duration) (exitCode int32, processStatus string, containerState string, reterr error) {
+	defer trace.End(trace.Begin(vc.ContainerID))
+
+	if vc == nil {
+		reterr = InternalServerError("Wait bad arguments")
+		return
+	}
+
+	// Get an API client to the portlayer
+	client := PortLayerClient()
+	if client == nil {
+		reterr = InternalServerError("Wait failed to create a portlayer client")
+		return
+	}
+
+	params := containers.NewContainerWaitParamsWithContext(ctx).
+		WithTimeout(int64(timeout.Seconds())).
+		WithID(vc.ContainerID)
+	results, err := client.Containers.ContainerWait(params)
+	if err != nil {
+		switch err := err.(type) {
+		case *containers.ContainerWaitNotFound:
+			// since the container wasn't found on the PL lets remove from the local
+			// cache
+			cache.ContainerCache().DeleteContainer(vc.ContainerID)
+			reterr = NotFoundError(vc.ContainerID)
+			return
+		case *containers.ContainerWaitInternalServerError:
+			reterr = InternalServerError(err.Payload.Message)
+			return
+		default:
+			reterr = InternalServerError(err.Error())
+			return
+		}
+	}
+
+	if results == nil || results.Payload == nil {
+		reterr = InternalServerError("Unexpected swagger error")
+	}
+
+	containerInfo := results.Payload
+
+	return *containerInfo.ProcessConfig.ExitCode, *containerInfo.ProcessConfig.Status, *containerInfo.ContainerConfig.State, nil
+}
+
+func (c *ContainerProxy) Signal(vc *viccontainer.VicContainer, sig uint64) error {
+	defer trace.End(trace.Begin(vc.ContainerID))
+
+	if vc == nil {
+		return InternalServerError("Signal bad arguments")
+	}
+
+	// Get an API client to the portlayer
+	client := PortLayerClient()
+	if client == nil {
+		return InternalServerError("Signal failed to create a portlayer client")
+	}
+
+	if running, err := c.ContainerRunning(vc); !running {
+		if err == nil {
+			return fmt.Errorf("Signal %s is not running", vc.ContainerID)
+		}
+	}
+
+	if sig == 0 || syscall.Signal(sig) == syscall.SIGKILL {
+		params := containers.NewContainerSignalParamsWithContext(ctx).WithID(vc.ContainerID).WithSignal(int64(syscall.SIGKILL))
+		if _, err := client.Containers.ContainerSignal(params); err != nil {
+			switch err.(type) {
+			case *containers.ContainerSignalNotFound:
+				return NotFoundError(vc.ContainerID)
+			case *containers.ContainerSignalInternalServerError:
+				// Container may have received the signal but we got an error.
+				// Give it some time and check again.
+				c.Wait(vc, killWaitForExit)
```

Comment:
I see your point, and I started to make the changes, but the problem is this, container.go and container_proxy.go isn't fully refactored yet.  I need to move the Stop() swagger code to container_proxy.go, and it's actually not a simple move.  It would make this PR overly large.  I want to do that in another PR.